lib/enigma_encoder.rb

```diff
@@ -1,3 +1,39 @@
 class EnigmaEncoder
 
+  def initialize
+    @whitelist = ('a'..'z').to_a << ' '
+  end
+
+  def keys(key)
+    key.split('').each_cons(2).map {|a,b| a + b}
+  end
+
+  def offset(date)
+    (date.to_i**2).to_s[-4..-1].split('')
+  end
+
+  def shifts(keys, offset)
+    keys.map.with_index do |key, index|
```

Comment:
could also do this with a zip:
`keys.zip(offset).map {|key, off| key.to_i + off.to_i }`

Reply:
oh thanks